src/vm/appdomain.cpp

```diff
@@ -12583,6 +12583,8 @@ AppDomain::RaiseAssemblyResolveEvent(
             if ( pSpec->IsIntrospectionOnly() 
 #ifdef FEATURE_FUSION
                     || pSpec->GetParentLoadContext() == LOADCTX_TYPE_UNKNOWN
+#elif FEATURE_CORECLR
+                    || pSpec->GetBindingContextFromParentAssembly(this) != GetTPABinderContext()
 #endif
```

Comment:
ResolveEventArgs.RequestingAssembly will be NULL for default context and will be the parent assembly for all other load context